### DISCUSSION TITLE
Dissect manager to use only individual parts?

### DISCUSSION THREAD

**Discussion by haarhoff-frs** [PiranhaCMS/piranha.core#2002] [General] [answered]
Hi,

I'm currently in the situation where I have to add blog functionality to a small internal tool. I don't really want to expose the whole Piranha manager to the users, I'd rather integrate the blog management with the current backend. I guess I'd have to mirror the controllers/pages in the manager implementation, and load the necessary Vue apps.

Would it be possible/feasible to render single sections from the manager to make i.e. the post-editing page and the media management page available inside the existing backend? Or is that a terrible idea? 

Next best thing would be to hide some parts of the manager, as I do not want users to be able to create new sites or change any settings. 

I guess it boils down to me asking whether this is something worth pursuing, or if I should just slab a WYSIWYG editor onto a razor page and call it a day 😅

**Comment by tidyui** [comment, marked as answer]
All of the different pages in the menu can be removed by changing item collection in `Piranha.Manager.Menu.Items`, this way you can remove all of the stuff you need from the manager. You could also achieve a similar experience by altering the permissions for the logged in user.

Given that both the page edit, and post edit view have a lot of dependencies on other components in the manager, like

* Media picker
* Tiny MCE
* Media preview
* Block selector
* Page/Post selector
* etc...

it's probably a lot of manual work just ripping out this part of the UI into another application.

Best regards

**Comment by haarhoff-frs** [comment]
Thanks for your suggestion. I'll use the manager as-is, using permissions to hide the items that should not be interacted with :)
